parsers/jscript.c

```diff
@@ -1877,9 +1949,99 @@ static bool parseStatement (tokenInfo *const token, tokenInfo *const parent, boo
 	deleteToken (method_body_token);
 	vStringDelete(saveScope);
 
+	JSCRIPT_DEBUG_LEAVE();
+
 	return is_terminated;
 }
 
+static bool parseES6Class(tokenInfo *const token,tokenInfo * const parent)
+{
+	JSCRIPT_DEBUG_ENTER();
+
+	tokenInfo * className = newToken ();
+	tokenInfo * base = NULL;
+
+	readToken(className);
+	// FIXME: Check if it's an identifier?
+
+	readToken(token);
+	if(isType(token,TOKEN_KEYWORD) && (token->keyword == KEYWORD_extends))
+	{
+		readToken(token);
+
+		if(isType(token,TOKEN_IDENTIFIER))
+		{
+			base = newToken();
+			copyToken(base,token,true);
+		}
+	}
+	
+	// emit class
+	// FIXME: Handle base?
+
+	JSCRIPT_DEBUG_PRINT("Emitting tag for class '%s'",vStringValue(className->string));
+
+	makeJsTag(className,JSTAG_CLASS,NULL);
+
+	if(base)
+		deleteToken(base);
+
+	while(!isType(token,TOKEN_OPEN_CURLY) && !isType(token,TOKEN_EOF))
```

Comment:
AFAICT this could reject a few more things to avoid dropping too much on partial/broken input, e.g. `;`